core/src/main/java/org/ehcache/core/EhcacheManager.java

```diff
@@ -654,9 +654,27 @@ public void destroyCache(final String alias) throws CachePersistenceException {
     if (alias == null) {
       throw new NullPointerException("Alias cannot be null");
     }
+
     LOGGER.debug("Destroying Cache '{}' in {}.", alias, simpleName);
-    removeAndCloseWithoutNotice(alias);
-    destroyPersistenceSpace(alias);
+
+    StatusTransitioner.Transition maintenance = null;
+    try {
+      maintenance = statusTransitioner.maintenance();
+      maintenance.succeeded();
+    } catch(IllegalStateException e) {
+      // the cache manager is already started, no need to put it in maintenance
```

Comment:
Agreed

On 8 September 2016 at 10:17, Louis Jacomet <notifications@github.com>
wrote:

> In core/src/main/java/org/ehcache/core/EhcacheManager.java
> <https://github.com/ehcache/ehcache3/pull/1421#discussion_r78014708>:
>
> >      LOGGER.debug("Destroying Cache '{}' in {}.", alias, simpleName);
> > -    removeAndCloseWithoutNotice(alias);
> > -    destroyPersistenceSpace(alias);
> > +
> > +    StatusTransitioner.Transition maintenance = null;
> > +    try {
> > +      maintenance = statusTransitioner.maintenance();
> > +      maintenance.succeeded();
> > +    } catch(IllegalStateException e) {
> > +      // the cache manager is already started, no need to put it in maintenance
>
> Correct - it just makes sure that if we failed to put in maintenance
> because it is already in maintenance in another thread we do not blindly
> proceed. Makes sense?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ehcache/ehcache3/pull/1421/files/abff3f27da04a6af37efa78e127ae9a16c052a8b#r78014708>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABKRM7Q668UTmqmfOkhHOrlDzpg_ABdvks5qoBkWgaJpZM4J3BDP>
> .
>